py/dml/c_backend.py

```diff
@@ -778,17 +778,22 @@ def generate_implement_method(device, ifacestruct, meth, indices):
         require_fully_typed(None, meth)
 
         # Calculate the expected method signature
-        ifacemethtype = ifacestruct.get_member_qualified(meth.name)
-        if not ifacemethtype:
+        member_type = ifacestruct.get_member_qualified(meth.name)
+        if not member_type:
             raise EMEMBER(meth.site, meth.parent.name, meth.name)
-        ifacemethtype = safe_realtype(ifacemethtype)
-        if isinstance(ifacemethtype, TPtr):
-            ifacemethtype = ifacemethtype.base
-        if not isinstance(ifacemethtype, TFunction):
-            raise EBTYPE(meth.site, ifacemethtype,
-                         TPtr(TFunction([], TVoid())))
-        iface_input_types = ifacemethtype.input_types[1:]
-        iface_num_outputs = 0 if ifacemethtype.output_type.void else 1
+        member_type = safe_realtype(member_type)
+        if not isinstance(member_type, TPtr):
+            raise EIMPLMEMBER(
+                meth.site,
+                f'{meth.parent.name}_interface_t.{meth.name}',
+                ifacestruct.declaration_site)
```

Comment:
Why did you decide to have the compatibility feature preserve the use of the thing in the context of interface objects (you haven't touched `mkInterfaceMethodRef`), but not here in the context of implements?

Reply:
Well spotted, I didn't think we had any code for this on the interface side, but indeed we do; `mkInterfaceMethodRef` indeed does
```
    if isinstance(ftype, TPtr):
        ftype = ftype.base
```
I will update that one too.

Reply:
What's the point of the compatibility feature if it doesn't support the most prominent existing use case of functions-in-structus?

Reply:
Ahh, I see.